gen/string.go

```diff
@@ -25,8 +25,19 @@ import (
 	"strings"
 	"unicode"
 	"unicode/utf8"
+
+	"go.uber.org/thriftrw/compile"
 )
 
+// ItemSpec is any Thrift compile items with a name and annotations.
+type ItemSpec interface {
```

Comment:
Perhaps just call this Spec and hoist it up somewhere common. This is an interface that captures all of the spec types.